src/main/java/com/minecolonies/coremod/colony/buildings/workerbuildings/BuildingNetherWorker.java

```diff
@@ -0,0 +1,334 @@
+                && itemStack.getItem() instanceof ArmorItem
+                && ((ArmorItem) itemStack.getItem()).getSlot() == EquipmentSlotType.HEAD, new Tuple<>(1, true));
+        keepX.put(itemStack -> !ItemStackUtils.isEmpty(itemStack)
+                && itemStack.getItem() instanceof ArmorItem
+                && ((ArmorItem) itemStack.getItem()).getSlot() == EquipmentSlotType.CHEST, new Tuple<>(1, true));
+        keepX.put(itemStack -> !ItemStackUtils.isEmpty(itemStack)
+                && itemStack.getItem() instanceof ArmorItem
+                && ((ArmorItem) itemStack.getItem()).getSlot() == EquipmentSlotType.LEGS, new Tuple<>(1, true));
+        keepX.put(itemStack -> !ItemStackUtils.isEmpty(itemStack)
+                && itemStack.getItem() instanceof ArmorItem
+                && ((ArmorItem) itemStack.getItem()).getSlot() == EquipmentSlotType.FEET, new Tuple<>(1, true));
+    }
+
+    @NotNull
+    @Override
+    public String getSchematicName()
+    {
+        return NETHER_WORKER;
+    }
+
+    @Override
+    public int getMaxBuildingLevel()
+    {
+        return CONST_DEFAULT_MAX_BUILDING_LEVEL;
+    }
+
+    /**
+     * Should the portal be closed on return? 
+     */
+    public boolean shallClosePortalOnReturn()
+    {
+        return getSetting(CLOSE_PORTAL).getValue();
+    }
+
+    @Override
+    public void onWakeUp()
+    {
+        super.onWakeUp();
+        snapTime = colony.getWorld().getDayTime();
+        if(this.currentPeriodDay < getPeriodDays())
+        {
+            this.currentPeriodDay++;
+        }
+        else
+        {
+            this.currentPeriodDay = 0;
+            this.currentTrips = 0;
+        }
+    }
+
+    @Override
+    public void deserializeNBT(final CompoundNBT compound)
+    {
+        super.deserializeNBT(compound);
+        if(compound.contains(TAG_CURRENT_TRIPS))
+        {
+            this.currentTrips = compound.getInt(TAG_CURRENT_TRIPS);
+        }
+
+        if(compound.contains(TAG_CURRENT_DAY))
+        {
+            this.currentPeriodDay = compound.getInt(TAG_CURRENT_DAY);
+        }
+    }
+
+    @Override
+    public CompoundNBT serializeNBT()
+    {
+        final CompoundNBT compound = super.serializeNBT();
+
+        compound.putInt(TAG_CURRENT_TRIPS, this.currentTrips);
+        compound.putInt(TAG_CURRENT_DAY, this.currentPeriodDay);
+
+        return compound;
+    }
+
+    @Override
+    public int buildingRequiresCertainAmountOfItem(final ItemStack stack, final List<ItemStorage> localAlreadyKept, final boolean inventory)
+    {
+        if (stack.isEmpty())
+        {
+            return 0;
+        }
+
+        if (inventory && getFirstModuleOccurance(MinimumStockModule.class).isStocked(stack))
+        {
+            return stack.getCount();
+        }
+
+        if (isAllowedFood(stack) && (localAlreadyKept.stream().filter(storage -> ISFOOD.test(storage.getItemStack())).mapToInt(ItemStorage::getAmount).sum() < STACKSIZE || !inventory))
```

Comment:
Well, the caveat with that is that you don't want farmers eating raw carrots they've just farmed, or guards eating zombie flesh, or rabbithands eating the carrots or the raw rabbit ... unless the restaurant is configured to hand out carrots, in which case maybe you do.

To repeat an idea that's been suggested before, it might be simplest if colonists had a special "food" inventory slot that only the player and restaurant can touch, and they'll only eat food from there.  It's slightly more silly in that the farmer with carrots would refuse to eat them until they dump them in the warehouse, go to the restaurant, and then are given carrots to eat, but that's probably only a short-term weirdness that will go away once the colony has better food supplies anyway.

A general food overhaul is probably out of scope for the nether miner PR anyway though 😁

Reply:
This code here applies only to the netherworker though. For the farmer stuff we got special job based checks.

Reply:
I know, but you were making general statements.  Though note that the nether miner can collect raw pork chops (and I think zombie flesh too)... (theoretically even cooked ones if they had a Fire Aspect sword, though I don't think that's considered in the simulated loot at present).

Reply:
"any edibles"

Reply:
And no, I was trying to make general statements about the netherminer